### ISSUE TITLE
Cannot import 'get_canonical_smiles' from 'baybe.utils'

### ISSUE THREAD

**Issue by mgreiner79** [emdgroup/baybe#120]
I am trying to run Baybe 0.7.2 with python 3.11 on a Windows machine.
I am trying out the example script from basic serialization (examples/Serialization/basic_serialization.py).
When trying to import Campaign, I get the following:
```Traceback (most recent call last):

  Cell In[2], line 1
    from baybe import Campaign

  File ~\code\auto_flow\.venv\Lib\site-packages\baybe\__init__.py:5
    from baybe.campaign import Campaign

  File ~\code\auto_flow\.venv\Lib\site-packages\baybe\campaign.py:13
    from baybe.parameters.base import Parameter

  File ~\code\auto_flow\.venv\Lib\site-packages\baybe\parameters\__init__.py:14
    from baybe.parameters.substance import SubstanceParameter

  File ~\code\auto_flow\.venv\Lib\site-packages\baybe\parameters\substance.py:13
    from baybe.utils import (

ImportError: cannot import name 'get_canonical_smiles' from 'baybe.utils' (C:\Users\M316675\code\auto_flow\.venv\Lib\site-packages\baybe\utils\__init__.py)
```

I had a look, and it seems that the problem is that substance.py explicitly imports `get_canonical_smiles`. However, in the ./utils/chemistry.py module, `get_canonical_smiles` is only defined if rdkit is installed. 
I don't have rdkit installed, so it fails when trying to import  'get_canonical_smiles' in substance.py.

I would recommend including rdkit in the dependencies for this project.

**Comment by Scienfitz** [comment]
Hi @mgreiner79 

RDKit and Mordred are optional dependencies of baybe that are only needed when using the `SubstanceParameter`. We made them optional since they are quite heavy and not all users need it.

Having said that, it seems that serialization example you mentioned uses this, but I think it probably shouldn't as it might be executed by users who only have the main dependencies installed. So good catch and thanks for this report!

you can install the needed deps via `pip install baybe[chem]`
For more info on optional dependencies please see this excerpt rom our README:
![image](https://github.com/emdgroup/baybe/assets/17951239/2983d0ba-f9b5-45f4-a2b6-50faad4ed9c9)

@AdrianSosic @AVHopp I think we can turn this issue into a PR on removing the `SubstanceParameter` from said example

**Comment by AdrianSosic** [comment]
Hi @mgreiner79,

Thanks for contacting us 🏆. Just to get everyone on the same page:
* The bad error message you observe was due to a bug in our package loading mechanism, which caused that a plain `baybe` install (i.e. without any optional dependencies) was not usable at all, since importing `baybe` automatically tried to import the `chem` stuff. This has already been fixed in #87 but the change has not yet been released. The problem unfortunately slipped through our tests since our `coretest` environment unintentionally installed the `chem` deps, which we also already fixed in #116.
* I will release a `0.7.3` probably later this afternoon or early next week which will fix all these problems for a regular pip-install. Until then, you can either do a git-install from `main` or explicitly install the package via `pip install "baybe[chem]"`
* With the new version, you will then get an appropriate error message when you attempt to use `chem`-related objects, such as the `SubstanceParameter`, i.e.
   ```python
   baybe.exceptions.OptionalImportError: The requested functionality requires the installation of optional chemistry dependencies. Please run "pip install 'baybe[chem]'"
  ```

Finally, thank you for pointing out specifically the problem with the serialization example. While we've planned an entire overhaul of all example files anyway, this specific one certainly should not require any optional dependencies, so I've just removed it (see #122) from the example code 👍🏼 

Let me know if you see any further problems. I'll leave this issue open for now and close it once I've released the new `0.7.3` version 🚀

**Comment by AdrianSosic** [comment]
Fixed in version `0.7.3`. Earlier versions require installing the `chem` dependencies, i.e. `pip install 'baybe[chem]'`
